### PR TITLE
Improve pachctl

### LINKED ISSUE #365: Make `pachctl` have better error messages when args are wrong
Right now if you do the wrong number of args to `pachctl` you get a very uninformative error message like:

```
Expected 1 args, got 0
```

We should make it so you get back `Usage` instead.

### PR DESCRIPTION
**Pull request by derekchiang**
Add block options to `InspectFile` and `ListCommit`

Fix #365

### CODE REVIEW

**Comment by derekchiang** [comment]
@jdoliner

File: src/server/pkg/cmd/cmd.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+		if len(args) < min || len(args) > max {
+			fmt.Printf("Expected %d to %d arguments, got %d.\n\n", min, max, len(args))
+			cmd.Usage()
+		} else {
+			if err := run(args); err != nil {
+				pkgcobra.ErrorAndExit("%v", err)
+			}
+		}
+	}
+}
```
Comment (jdoliner) [inline src/server/pkg/cmd/cmd.go]: I feel like having a dependency on pkgcobra for `ErrorAndExit` is a little bit silly. We can just printf the error and then call os.Exist ourselves.

File: src/server/pps/cmds/cmds.go [hunk 2]
```diff
@@ -113,13 +114,14 @@ The increase the throughput of a job increase the Shard paremeter.
 			return writer.Flush()
 		}),
 	}
+	inspectJob.Flags().BoolVarP(&block, "block", "b", false, "block until the job has either succeeded or failed")
 
 	var pipelineName string
 	listJob := &cobra.Command{
```
Comment (jdoliner) [inline src/server/pps/cmds/cmds.go]: Could we add `InputCommit` as an option here.

File: src/server/pps/cmds/cmds.go [hunk 3]
```diff
@@ -288,7 +290,7 @@ All jobs created by a pipeline will create commits in the pipeline's repo.
 		Use:   "run-pipeline pipeline-name [-f job.json]",
 		Short: "Run a pipeline once.",
 		Long:  fmt.Sprintf("Run a pipeline once, optionally overriding some pipeline options by providing a spec.  The spec looks like this:\n%s", exampleRunPipelineSpec),
```
Comment (jdoliner) [inline src/server/pps/cmds/cmds.go]: I think it would be good to include the extend pipeline spec docs that we have on GH in this help string.

**Comment by jdoliner** [comment]
@sjezewski did a few special things in the `mount` cmd to make usage show up when the command errors. I think given the new `cmd` package that's now redundant and should be migrated to just using `RunFixedArgs`. Let's do that as part of this PR too.

**Comment by sjezewski** [comment]
Yup, good call. Building now:

https://travis-ci.org/pachyderm/pachyderm/builds/127673513

File: src/server/pkg/cmd/cmd.go (hunk 1 above)
Comment (jdoliner) [inline src/server/pkg/cmd/cmd.go]: Given that we're bumping up against deadlines I don't feel too strongly about this one. I'm fine merging w/o it.

**Comment by derekchiang** [comment]
@jdoliner comments addressed.  Notably, `pachctl` now directly reads from `doc/pipeline_spec.md` to include the entire thing in the help string for `create-pipeline`.  Wanna take another look?

**Comment by jdoliner** [comment]
Nice LGTM

**Pull request merged**